### PR TITLE
Update docs with GA version of stream and task apps/starters

### LINKED ISSUE #1438: References to app starter stream and task URLs are outdated.
We should reference the new GA releases of stream and task.

Also, need to decide the path to getting off of using bit.ly as a URL shortener since it doesn't support https access.

We can host URLs of the form `https://cloud.spring.io/spring-cloud-stream-app-starters/releases/Avogadro.RELEASE/stream-apps-kafka-10-docker` which isn't particularly short...

We can also host our own app on PWS... use `http://stream-app-starters.cfapps.io/Bacon-RELEASE-stream-applications-rabbit-maven` format

### PR DESCRIPTION
**Pull request by sobychacko**
Fix typos in some links
Update spring-cloud-stream-app-starter gh link in analytics section

Fixes #1438

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
LGTM.

@sobychacko I'd like to know the plan for this suggestions from #1438 :

> We can host URLs of the form https://cloud.spring.io/spring-cloud-stream-app-starters/releases/Avogadro.RELEASE/stream-apps-kafka-10-docker which isn't particularly short...

> We can also host our own app on PWS... use http://stream-app-starters.cfapps.io/Bacon-RELEASE-stream-applications-rabbit-maven format

**Comment by sobychacko** [comment]
@ilayaperumalg I don't think we reached a decision on that yet. Not sure, will make a decision before the GA timeline. Thus, going with bit.ly links once again.

**Comment by markpollack** [comment]
We are going with bit.ly for now.

File: spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc [hunk 1]
```diff
@@ -54,9 +54,9 @@ expressed in a properties style:
 
 For example:
 
-`task.timestamp=maven://org.springframework.cloud.task.app:timestamp-task:1.0.0.BUILD-SNAPSHOT`
+`task.timestamp=maven://org.springframework.cloud.task.app:timestamp-task:1.2.1.BUILD-SNAPSHOT`
```
Comment (markpollack) [inline spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc]: should reference known GA versions.

File: spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc [hunk 2]
```diff
@@ -54,9 +54,9 @@ expressed in a properties style:
...
 
-`processor.transform=maven://org.springframework.cloud.stream.app:transform-processor-rabbit:1.0.3.BUILD-SNAPSHOT`
+`processor.transform=maven://org.springframework.cloud.stream.app:transform-processor-rabbit:1.2.1.BUILD-SNAPSHOT`
```
Comment (markpollack) [inline spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc]: should reference known GA versions

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 3]
```diff
@@ -59,8 +59,8 @@ For example, if you would like to register the snapshot versions of the `http` a
 applications built with the RabbitMQ binder, you could do the following:
 
 ```
-dataflow:>app register --name http --type source --uri maven://org.springframework.cloud.stream.app:http-source-rabbit:1.1.2.BUILD-SNAPSHOT
-dataflow:>app register --name log --type sink --uri maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.1.2.BUILD-SNAPSHOT
+dataflow:>app register --name http --type source --uri maven://org.springframework.cloud.stream.app:http-source-rabbit:1.2.1.BUILD-SNAPSHOT
```
Comment (markpollack) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: should reference know GA versions

Comment (sobychacko) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: but this sections refers to the snapshot version of the apps.

**Comment by markpollack** [comment]
merged with changes to make some snapshot imports reference GA versions.

bb1674139464d409601a6afec6123c34f6237341

**Pull request closed without merging**